dht.go

```diff
@@ -94,31 +109,61 @@ var (
 
 // New creates a new DHT with the specified host and options.
 func New(ctx context.Context, h host.Host, options ...opts.Option) (*IpfsDHT, error) {
-	var cfg opts.Options
+	cfg := &opts.Options{}
 	cfg.BucketSize = KValue
 	if err := cfg.Apply(append([]opts.Option{opts.Defaults}, options...)...); err != nil {
 		return nil, err
 	}
-	dht := makeDHT(ctx, h, cfg.Datastore, cfg.Protocols, cfg.BucketSize)
-	dht.autoRefresh = cfg.RoutingTable.AutoRefresh
-	dht.rtRefreshPeriod = cfg.RoutingTable.RefreshPeriod
-	dht.rtRefreshQueryTimeout = cfg.RoutingTable.RefreshQueryTimeout
 
-	dht.maxRecordAge = cfg.MaxRecordAge
-	dht.enableProviders = cfg.EnableProviders
-	dht.enableValues = cfg.EnableValues
+	// set seedsProposer, snapshotter & fallback peers if not set
+	if cfg.Persistence.SeedsProposer == nil {
+		cfg.Persistence.SeedsProposer = persist.NewRandomSeedsProposer()
+	}
+	snapshotter := cfg.Persistence.Snapshotter
+	if snapshotter == nil {
+		s, err := persist.NewDatastoreSnapshotter(cfg.Datastore, persist.DefaultSnapshotNS)
+		// should never happen
+		if err != nil {
+			logger.Errorf("failed to initialize the default datastore backed snapshotter, err: %s", err)
+			return nil, pkgerr.WithMessage(err, "failed to initialize the default datastore backed snapshotter")
+		}
+		snapshotter = s
+	}
 
-	// register for network notifs.
-	dht.host.Network().Notify((*netNotifiee)(dht))
+	if len(cfg.Persistence.FallbackPeers) == 0 {
+		cfg.Persistence.FallbackPeers = getDefaultBootstrapPeerIDs()
+	}
```

Comment:
we can move this to `Defaults` as well?

Reply:
We don't want the opts package to depend on the dht package and the opts package isn't the right place to put the function definition.

Reply:
k, I don't feel too strongly about this. Just like keeping all the defaults in one place when we can so they're easier to find 😄.